test/test_reynir.py

```diff
@@ -317,13 +317,134 @@ def test_auto_uppercase():
     s = g.parse_single("hver er guðni th jóhannesson")
     assert detokenize(s.tokens) == "hver er Guðni Th Jóhannesson"
     assert "Guðni Th Jóhannesson" in s.tree.persons
+
+    s = g.parse_single("hver er guðni th. jóhannesson")
+    assert detokenize(s.tokens) == "hver er Guðni Th. Jóhannesson"
+    assert "Guðni Th. Jóhannesson" in s.tree.persons
+
     s = g.parse_single("hver er gunnar thoroddsen")
     assert detokenize(s.tokens) == "hver er Gunnar Thoroddsen"
     assert "Gunnar Thoroddsen" in s.tree.persons
+
     s = g.parse_single("hver er eliza reid")
     assert detokenize(s.tokens) == "hver er Eliza Reid"
     assert "Eliza Reid" in s.tree.persons
 
+    s = g.parse_single("hver er hæð jóns")
+    assert detokenize(s.tokens) == "hver er hæð Jóns"
+    assert "Jón" in s.tree.persons
+
+    s = g.parse_single("hver er hæð sólar í dag í reykjavík")
+    assert "Í" not in detokenize(s.tokens)
+    assert "Sólar Í Dag Í Reykjavík" not in s.tree.persons
```

Comment:
Could add further tests for handling of "sólar" and "dag", I know some of them might not be producing 100% correct results but at least having them here and eventually commented out would be good as a reminder.